tracer/src/Datadog.Trace.Tools.dd_dotnet/CreatedumpCommand.cs

```diff
@@ -0,0 +1,676 @@
+// <copyright file="CreatedumpCommand.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Configuration;
+using Microsoft.Diagnostics.Runtime;
+using Spectre.Console;
+
+namespace Datadog.Trace.Tools.dd_dotnet;
+
+internal class CreatedumpCommand : Command
+{
+    private const int MethodNameMaxLength = 1024;
+
+    private static readonly List<string> Errors = new();
+    private static ClrRuntime? _runtime;
+
+    private readonly Argument<string[]> _allArguments = new("args");
+
+    public CreatedumpCommand()
+        : base("createdump")
+    {
+        // Use a string[] argument to match everything
+        // We want to be able to forward the command to createdump even when we don't understand it
+        AddArgument(_allArguments);
+        IsHidden = true;
+
+        this.SetHandler(Execute);
+    }
+
+    [UnmanagedCallersOnly]
+    private static unsafe int ResolveManagedMethod(IntPtr ip, ResolveMethodData* methodData)
+    {
+        try
+        {
+            if (_runtime == null)
+            {
+                Errors.Add("ClrRuntime is not initialized");
+                return -2;
+            }
+
+            var method = _runtime.GetMethodByInstructionPointer((ulong)ip);
+
+            if (method == null)
+            {
+                return 1;
+            }
+
+            methodData->SymbolAddress = method.NativeCode;
+            methodData->ModuleAddress = method!.Type!.Module!.ImageBase;
```

Comment:
Nice catch. Yeah I added those yesterday because for my tests I switched to an older version of ClrMD that didn't properly declare the nullable annotations. Gonna fix it.

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/5451/commits/ee073bd107bcc3cebfe47566ce399d2a5944e2a9